otx/core/data/adapter/base_dataset_adapter.py

```diff
@@ -0,0 +1,250 @@
+
+    @abstractmethod
+    def get_otx_dataset(self) -> DatasetEntity:
+        """Get DatasetEntity.
+
+        Args:
+            datumaro_dataset (dict): A Dictionary that includes subset dataset(DatasetEntity)
+
+        Returns:
+            DatasetEntity:
+        """
+        raise NotImplementedError
+
+    def get_label_schema(self) -> LabelSchemaEntity:
+        """Get Label Schema."""
+        return self._generate_default_label_schema(self.label_entities)
+
+    def _detect_dataset_format(self, path: str) -> str:
+        """Detect dataset format (ImageNet, COCO, ...)."""
+        return datumaro.Environment().detect_dataset(path=path)
+
+    def _generate_empty_label_entity(self) -> LabelGroup:
+        """Generate Empty Label Group for H-label, Multi-label Classification."""
+        empty_label = LabelEntity(name="Empty label", is_empty=True, domain=self.domain)
+        empty_group = LabelGroup(name="empty", labels=[empty_label], group_type=LabelGroupType.EMPTY_LABEL)
+        return empty_group
+
+    def _generate_default_label_schema(self, label_entities: list) -> LabelSchemaEntity:
+        """Generate Default Label Schema for Multi-class Classification, Detecion, Etc."""
+        label_schema = LabelSchemaEntity()
+        main_group = LabelGroup(
+            name="labels",
+            labels=label_entities,
+            group_type=LabelGroupType.EXCLUSIVE,
+        )
+        label_schema.add_group(main_group)
+        return label_schema
+
+    def _prepare_label_information(
+        self,
+        datumaro_dataset: dict,
+    ) -> dict:
+        # Get datumaro category information
+        if self.is_train_phase:
+            label_categories_list = (
+                datumaro_dataset[Subset.TRAINING].categories().get(DatumaroAnnotationType.label, None)
+            )
+        else:
+            label_categories_list = (
+                datumaro_dataset[Subset.TESTING].categories().get(DatumaroAnnotationType.label, None)
+            )
+        category_items = label_categories_list.items
+        label_groups = label_categories_list.label_groups
+
+        # LabelEntities
+        label_entities = [
+            LabelEntity(name=class_name.name, domain=self.domain, is_empty=False, id=ID(i))
+            for i, class_name in enumerate(category_items)
+        ]
+
+        return {"category_items": category_items, "label_groups": label_groups, "label_entities": label_entities}
+
+    def _select_data_type(self, data_candidates: Union[list, str]) -> str:
+        """Select specific type among candidates.
+
+        Args:
+            data_candidates (list): Type candidates made by Datumaro.Environment().detect_dataset()
+
+        Returns:
+            str: Selected data type
+        """
+        return data_candidates[0]
+
+    def _get_ann_scene_entity(self, shapes: List[Annotation]) -> AnnotationSceneEntity:
+        annotation_scene = None  # type: Union[NullAnnotationSceneEntity, AnnotationSceneEntity, None]
```

Comment:
maybe we could write this as:
```suggestion
        annotation_scene: Optional[AnnotationSceneEntity] = None
```